### PR TITLE
.collapse('hide') on hidden uninit-ed collapsible no longer shows it

### LINKED ISSUE #15315: .collapse('hide') on a hidden uninitialized collapsible shows it
Calling `collapse('hide')` on a collapse panel that is collapsed and has not yet been initialized will toggle the panel (which starts the transition to being shown) when it creates the new Collapse object. Then the call to `data[option]()` won't hide the panel because it is  still transitioning.

### PR DESCRIPTION
**Pull request by cvrebert**
Fixes #15315.
Closes #15807.
X-Ref: #14282, #14417

CC: @XhmikosR @hnrch02 @juthilo for review

### CODE REVIEW

**Comment by cvrebert** [comment]
Damn. The one old test this makes fail [dates back to the advent of the Collapse plugin's test suite](https://github.com/twbs/bootstrap/commit/1ef5fa7d6b4e50230c0c12919b0a06a9a2ac07f1#diff-4773e851610b64b333c3075ef5c248b0R22), so we might need to ask @fat about this.
I'm not sure why it's testing for adding `style="height: 0px;"`, considering that `.collapse` without `.in` sets `display: none` anyway...

File: js/tests/unit/collapse.js [hunk 1]
```diff
@@ -177,6 +177,34 @@ $(function () {
...
+      .appendTo('#qunit-fixture')
+      .on('show.bs.collapse', function () {
+        assert.ok(false, 'showing a previously-uninitialized hidden collapse when "hide" method called')
+      })
+      .bootstrapCollapse('hide')
+
+    setTimeout(done, 0)
+  })
+
+  QUnit.test('should hide a collapse when initialized with "hide" if not already hidden', function (assert) {
```
Comment (hnrch02) [inline js/tests/unit/collapse.js]: Maybe change the name to "should hide a collapse when initialized with "hide" *option* if not already hidden"

File: js/tests/unit/collapse.js [hunk 2]
```diff
@@ -177,6 +177,34 @@ $(function () {
...
+    assert.expect(1)
+    var done = assert.async()
+
+    $('<div class="collapse in"></div>')
```
Comment (hnrch02) [inline js/tests/unit/collapse.js]: $('<div class="collapse in"/>')

File: js/tests/unit/collapse.js [hunk 3]
```diff
@@ -177,6 +177,34 @@ $(function () {
...
   })
 
+  QUnit.test('should not show a collapse when initialized with "hide" if already hidden', function (assert) {
+    assert.expect(0)
+    var done = assert.async()
+
+    $('<div class="collapse"></div>')
+      .appendTo('#qunit-fixture')
+      .on('show.bs.collapse', function () {
+        assert.ok(false, 'showing a previously-uninitialized hidden collapse when "hide" method called')
```
Comment (hnrch02) [inline js/tests/unit/collapse.js]: Again, not a native speak but shouldn't it be "showing a previously-uninitialized hidden collapse when "hide" method *is* called"

**Comment by hnrch02** [comment]
I think it's save to remove/rework that old test.

File: js/tests/unit/collapse.js (hunk 2 above)
Comment (cvrebert) [inline js/tests/unit/collapse.js]: That's not valid HTML and such shortcuts have caused some problems in the past (I think it with IE8, but still).

File: js/tests/unit/collapse.js (hunk 1 above)
Comment (cvrebert) [inline js/tests/unit/collapse.js]: I'm basically just following how the analogous "show" tests above these are phrased

File: js/tests/unit/collapse.js (hunk 2 above)
Comment (hnrch02) [inline js/tests/unit/collapse.js]: `grep "/>" js/tests/unit/*` and https://github.com/twbs/bootstrap/pull/13853#discussion_r14574124

File: js/tests/unit/collapse.js (hunk 1 above)
Comment (hnrch02) [inline js/tests/unit/collapse.js]: Okay.

File: js/tests/unit/collapse.js (hunk 3 above)
Comment (kkirsche) [inline js/tests/unit/collapse.js]: I would agree with that @hnrch02

File: js/tests/unit/collapse.js (hunk 2 above)
Comment (hnrch02) [inline js/tests/unit/collapse.js]: We can change it for the whole suite if you feel strongly about it. Can't remember any problem with IE8 though.

File: js/tests/unit/collapse.js (hunk 1 above)
Comment (cvrebert) [inline js/tests/unit/collapse.js]: Done.

File: js/tests/unit/collapse.js (hunk 3 above)
Comment (cvrebert) [inline js/tests/unit/collapse.js]: Rephrased

**Comment by cvrebert** [comment]
Ok, took the plunge and removed that one old assertion.

**Comment by cvrebert** [comment]
@XhmikosR LGTY?

**Comment by XhmikosR** [comment]
Makes sense. :+1:

**Pull request merged**